hasher-matcher-actioner/hmalib/common/actioner_models.py

```diff
@@ -43,49 +45,115 @@ class ActionRule:
-        return get(url)
+class WebhookGetActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
+        return get(self.url)
 
 
-class Put(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
+class WebhookPutActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
         data = kwargs.get("data")
-        return put(url, data)
+        return put(self.url, data)
 
 
-class Delete(HTTPRequest):
-    @staticmethod
-    def call(url: TUrl, **kwargs) -> Response:
-        return delete(url)
+class WebhookDeleteActionPerformer(WebhookActionPerformer):
+    def call(self, **kwargs) -> Response:
+        return delete(self.url)
 
 
-class ActionPerformer:
-    def perform_action(self, match_message: MatchMessage) -> None:
-        raise NotImplementedError()
+def get_all_subclasses_rec(recursive: t.Set[t.Type]) -> t.Set[t.Type]:
+    subclasses = {subclass for cls in recursive for subclass in cls.__subclasses__()}
+    union = recursive.union(subclasses)
+    if union != recursive:
+        return get_all_subclasses_rec(union)
+    return recursive
 
 
 @dataclass
-class WebhookActionPerformer(ActionPerformer):
-
-    request_type: t.Type[HTTPRequest]
-    url: TUrl
-
-    def perform_action(self, match_message: MatchMessage) -> None:
-        kwargs = {"url": self.url, "data": match_message.to_sns_message()}
-        self.request_type.call(**kwargs)
+class ActionPerformerConfig(config.HMAConfig):
+    """
+    An adapter to use the HMAConfig Class to store differnet types of
+    ActionPerformer configs. Since ActionPerformer configs are all unique by label but may
+    be of different types, we store the name of the concrete type and the
+    attributes of that type in AWS.
+
+    DO NOT CALL config.HMAConfig FUNCTIONS DIRECTLY
+
+    To load a config from a label use ActionPerformerConfig.get_performer
+
+    To update an action performer ActionPerformerConfig.update_performer
+    """
+
+    concrete_type: str
+    concrete_type_attrs: t.Dict[str, str]
+
+    @classmethod
+    def get_performer(cls, action_label: ActionLabel) -> t.Optional["ActionPerformer"]:
+        """
+        Load from AWS the correct actioner config as the correct concrete class if it exists
+        """
+        name = action_label.value
+        performer_config: t.Optional[ActionPerformerConfig] = super().get(name)
+        if performer_config:
+            subclasses_classes = {
+                cls.__name__: cls for cls in get_all_subclasses_rec({ActionPerformer})
+            }
+            return subclasses_classes[performer_config.concrete_type](
+                action_label=action_label, **performer_config.concrete_type_attrs
+            )
+        return None
+
+    @classmethod
+    def update_performer(cls, action_performer: ActionPerformer) -> None:
+        """
+        Use this function to update the config in AWS instead of config.update_config directly
```

Comment:
Hmm, this somewhat defeats the purpose of having a standard way of updating the configs if the first ever instance doesn't use it.